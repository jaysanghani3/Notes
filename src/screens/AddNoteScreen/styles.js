import {StyleSheet} from 'react-native';
import Colors from '../../utils/Colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 14,
    justifyContent: 'space-between',
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    fontSize: 25,
    fontWeight: 'bold',
    padding: 5,
    borderBottomWidth: 1,
    borderColor: Colors.gray,
  },
  detailsInput: {
    fontSize: 17,
    padding: 10,
    borderWidth: 1,
    borderColor: Colors.gray,
    borderRadius: 10,
    marginVertical: 10,
    maxHeight: 250,
  },
  btnGrp: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  btn: {
    backgroundColor: Colors.violet,
    padding: 15,
    borderRadius: 8,
    marginBottom: 5,
    width: '48%',
  },
  btnFont: {
    textAlign: 'center',
    color: Colors.white,
    fontWeight: 'bold',
  },
  noteView: {
    flex: 1,
    backgroundColor: Colors.white,
    padding: 20,
    borderRadius: 8,
  },
  noteViewTitle: {
    fontSize: 35,
    fontWeight: 'bold',
    color: Colors.violet,
    borderBottomColor: Colors.gray,
    borderBottomWidth: 1,
    paddingBottom: 10,
  },
  noteViewDetails: {
    fontSize: 16,
    color: Colors.gray,
    marginTop: 30,
  },
  titleCount: {
    textAlign: 'right',
    color: Colors.gray,
  },
  maxCount: {
    textAlign: 'right',
    color: Colors.error,
  },
});

export default styles;
