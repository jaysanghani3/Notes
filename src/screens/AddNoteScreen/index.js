import {
  ScrollView,
  Text,
  TextInput,
  TouchableOpacity,
  View,
  Keyboard,
} from 'react-native';
import React, {useEffect, useState} from 'react';
import styles from './styles';
import NetInfo from '@react-native-community/netinfo';
import Loader from '../../components/Loader';
import globalStyles from '../../utils/globalStyles';
import Strings from '../../translation/Strings';
import {
  addNotes,
  updateNote,
  updateNoteOffline,
  addNoteOffline,
} from '../../services/firestore';

const AddNoteScreen = ({route, navigation}) => {
  const {noteTitle, noteDetails, noteId, screen} = route.params;
  const [title, setTitle] = useState(noteTitle);
  const [details, setDetails] = useState(noteDetails);
  const [isLoading, setIsLoading] = useState(false);
  const [isConnected, setIsConnected] = useState(true);
  const [btnText, setBtnText] = useState(Strings.addBtn);
  const [titleLength, setTitleLength] = useState(title.length);
  const [titleError, setTitleError] = useState('');

  useEffect(() => {
    const unsubscribe = NetInfo.addEventListener(state => {
      setIsConnected(state.isConnected);
    });
    return () => {
      unsubscribe();
    };
  }, []);

  useEffect(() => {
    if (screen === Strings.addScreenTitle) {
      navigation.setOptions({title: Strings.addScreenTitle});
      setBtnText(Strings.addBtn);
    } else if (screen === Strings.editScreenTitle) {
      navigation.setOptions({title: Strings.editScreenTitle});
      setBtnText(Strings.updateBtn);
    } else if (screen === Strings.viewScreenTitle) {
      navigation.setOptions({title: Strings.viewScreenTitle});
    }
  }, [navigation]);

  const handleOfflineSupport = async () => {
    try {
      navigation.goBack();
      if (screen === Strings.editScreenTitle) {
        updateNoteOffline(noteId, title, details);
      } else if (screen === Strings.addScreenTitle) {
        addNoteOffline(title, details);
      }
      setIsLoading(false);
    } catch (e) {
      console.log(e);
    }
  };

  const handleSave = async () => {
    Keyboard.dismiss();
    if (!title && !details) {
      setTitleError(Strings.required);
    } else if (!title) {
      setTitleError(Strings.required);
    } else if (!details) {
    } else {
      try {
        if (!isConnected) {
          handleOfflineSupport();
        } else {
          const newNoteId = Math.random().toString(36).substring(7);
          if (screen === Strings.editScreenTitle) {
            updateNote(noteId, title, details);
          } else if (screen === Strings.addScreenTitle) {
            addNotes(newNoteId, title, details);
          }
          navigation.goBack();
        }
      } catch (e) {
        console.log(e);
      }
    }
  };

  const handleTitleChange = title => {
    setTitle(title);
    setTitleLength(title.length);
    title.length >= Strings.titleLength
      ? setTitleError(Strings.titleLengthError)
      : setTitleError('');
  };

  const handleDetailsChange = details => {
    setDetails(details);
  };

  return (
    <>
      {screen === Strings.viewScreenTitle ? (
        <View style={styles.noteView}>
          <Text style={styles.noteViewTitle}>{noteTitle}</Text>
          <ScrollView>
            <Text style={styles.noteViewDetails}>{noteDetails}</Text>
          </ScrollView>
        </View>
      ) : (
        <View style={styles.container}>
          <Loader state={isLoading} />

          <View>
            <TextInput
              placeholder={Strings.titlePlaceholder}
              maxLength={Strings.titleLength}
              style={styles.title}
              defaultValue={title}
              onChangeText={title => {
                handleTitleChange(title);
              }}
            />
            <View style={styles.row}>
              <Text style={globalStyles.error}>{titleError}</Text>
              <Text
                style={
                  titleLength === Strings.titleLength
                    ? styles.maxCount
                    : styles.titleCount
                }>
                {titleLength}/{Strings.titleLength}
              </Text>
            </View>
            <ScrollView>
              <TextInput
                placeholder={Strings.detailsPlaceholder}
                style={styles.detailsInput}
                multiline={true}
                numberOfLines={10}
                textAlignVertical="top"
                defaultValue={details}
                onChangeText={details => {
                  handleDetailsChange(details);
                }}
              />
            </ScrollView>
          </View>
          <View style={styles.btnGrp}>
            <TouchableOpacity
              style={styles.btn}
              onPress={() => {
                setDetails('');
                setTitle('');
              }}>
              <Text style={styles.btnFont}>{Strings.resetBtn}</Text>
            </TouchableOpacity>
            <TouchableOpacity style={styles.btn} onPress={handleSave}>
              <Text style={styles.btnFont}>{btnText}</Text>
            </TouchableOpacity>
          </View>
        </View>
      )}
    </>
  );
};
export default AddNoteScreen;
