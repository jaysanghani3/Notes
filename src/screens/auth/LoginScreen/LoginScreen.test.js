import AddNoteScreen from './index';
import React from 'react';
import renderer from 'react-test-renderer';
import {test, expect} from '@jest/globals';

test('renders correctly', () => {
  const tree = renderer.create(<AddNoteScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
