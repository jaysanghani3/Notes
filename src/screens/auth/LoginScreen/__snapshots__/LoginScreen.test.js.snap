// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 25,
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#000000",
          "fontSize": 30,
          "fontWeight": "bold",
        }
      }
    >
      Welcome
    </Text>
    <Text>
      Enter your details to log in.
    </Text>
  </View>
  <View
    style={
      {
        "marginVertical": 40,
      }
    }
  >
    <View
      style={
        {
          "marginVertical": 10,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderColor": "#B6B6B6",
            "borderRadius": 8,
            "borderWidth": 1,
            "display": "block",
            "flexDirection": "row",
            "paddingStart": 10,
          }
        }
      >
        <TextInput
          defaultValue=""
          inputMode="email"
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="#B6B6B6"
          style={
            {
              "flex": 1,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#D32F2F",
          }
        }
      />
    </View>
    <View
      style={
        {
          "marginVertical": 10,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#ffffff",
            "borderColor": "#B6B6B6",
            "borderRadius": 8,
            "borderWidth": 1,
            "display": "block",
            "flexDirection": "row",
            "paddingStart": 10,
          }
        }
      >
        <TextInput
          defaultValue=""
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            {
              "flex": 1,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "padding": 20,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/show.png",
              }
            }
            style={
              {
                "height": 25,
                "position": "absolute",
                "right": 15,
                "top": 11,
                "width": 25,
              }
            }
          />
        </View>
      </View>
      <Text
        style={
          {
            "color": "#D32F2F",
          }
        }
      />
    </View>
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#6574FF",
          "textAlign": "right",
        }
      }
    >
      Forgot Password
    </Text>
  </View>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#6574FF",
          "borderRadius": 8,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#B6B6B6",
        "borderTopWidth": 1,
        "marginTop": 50,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "textAlign": "center",
        }
      }
    >
      Don't have an account? 
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#6574FF",
          }
        }
      >
        Sign Up
      </Text>
    </Text>
  </View>
</View>
`;
